src/include/homestore/replication/repl_dev.h

```diff
@@ -367,6 +381,10 @@ class ReplDevListener {
     /// @brief Free up user-defined context inside the snapshot_obj that is allocated during read_snapshot_obj.
     virtual void free_user_snp_ctx(void*& user_snp_ctx) = 0;
 
+    /// @brief some repl_dev event might be handle by upper layer. This is where homestore gets the callbacks for these
+    /// events.
+    virtual void* get_event_handler(event_type_t event_type) { return nullptr; }
```

Comment:
This is anti-pattern with existing CB.   Suggesting follow the member function pattern for all the CB . If giving upper layer a choice not to implement such CB is important,  we can make it non-virtual.

Reply:
I have change the interfaces ， ptal